InventoryTweaks/Patches/Slot_RemoveItem_MultiPatch.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Reflection;
+using HarmonyLib;
+
+// ReSharper disable InconsistentNaming
+
+namespace thmsn.InventoryTweaks.Patches
+{
+    /// <summary>
+    ///     Checks several methods where <see cref="Slot.RemoveItem" /> is used for changes to selected hotbar slot
+    /// </summary>
+    public static class Slot_RemoveItem_MultiPatch
+    {
+        private static readonly AccessTools.FieldRef<BlockCreator, Network_Player> BlockCreator_playerNetwork =
+            AccessTools.FieldRefAccess<BlockCreator, Network_Player>(
+                AccessTools.Field(typeof(BlockCreator), "playerNetwork"));
+
+        //
+        // Targets
+        //
+        private static readonly MethodInfo PlayerInventory_RemoveSelectedHotSlotItem_Target =
+            AccessTools.Method(typeof(PlayerInventory), nameof(PlayerInventory.RemoveSelectedHotSlotItem));
+
+        private static readonly MethodInfo Inventory_RemoveItem_Target =
+            AccessTools.Method(typeof(Inventory), nameof(Inventory.RemoveItem));
+
+        private static readonly MethodInfo BlockCreator_CreateBlock_Target =
+            AccessTools.Method(typeof(BlockCreator), nameof(BlockCreator.CreateBlock));
+
+        //
+        // Patches
+        //
+        private static readonly HarmonyMethod commonPrefix =
+            new HarmonyMethod(typeof(Slot_RemoveItem_MultiPatch), nameof(CommonPrefix));
+
+        private static readonly HarmonyMethod commonPostfix =
+            new HarmonyMethod(typeof(Slot_RemoveItem_MultiPatch), nameof(CommonPostfix));
+
+        private static readonly HarmonyMethod inventory_RemoveItem_Prefix =
+            new HarmonyMethod(typeof(Slot_RemoveItem_MultiPatch), nameof(Inventory_RemoveItem_Prefix));
+
+        private static readonly HarmonyMethod blockCreator_CreateBlock_Prefix =
+            new HarmonyMethod(typeof(Slot_RemoveItem_MultiPatch), nameof(BlockCreator_CreateBlock_Prefix));
+
+        internal static void Patch(Harmony harmony)
+        {
+            harmony.Patch(PlayerInventory_RemoveSelectedHotSlotItem_Target, commonPrefix, commonPostfix);

```

Comment:
I don't have the raft code on my backbone, but I assume this is patched for when for example your hook is breaking and it needs to be replaced with a new one?

Reply:
Added comment in new edition
Battery insertion, tank/purifier collection with empty water holder and shark bait throwing
```
Battery.Insert
Block_CookingStand_Purifier.CollectItem
SharkBaitThrower.OnHitWater
Tank.CollectItem
```